snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -215,38 +225,33 @@ def _migrate_state_file(self):
             if step:
                 os.remove(self.plugin.statedir)
                 os.makedirs(self.plugin.statedir)
-                self.mark_done(step)
+                self.mark_done(steps.Step(step))
 
     def notify_part_progress(self, progress, hint=''):
         logger.info('%s %s %s', progress, self.name, hint)
 
-    def last_step(self):
-        for step in reversed(common.COMMAND_ORDER):
+    def latest_step(self):
+        for step in reversed(steps.STEPS):
             if os.path.exists(
                     states.get_step_state_file(self.plugin.statedir, step)):
                 return step
 
-        return None
+        raise errors.NoLatestStepError(self.name)
 
     def next_step(self):
-        next_step = None
-        for step in reversed(common.COMMAND_ORDER):
-            if os.path.exists(
-                    states.get_step_state_file(self.plugin.statedir, step)):
-                break
-            next_step = step
-
-        return next_step
+        latest_step = None
```

Comment:
if what we return is the next_step, should what we eval be the current_step instead of the latest_step? I am not sure why, but calling this latest_step confuses me, we had this conversation already as this is triggering things (I think we even opened a bug for this).

Reply:
Ah, everything revolves around `latest_step` now it seems

Reply:
Nothing has changed here, it was just called "last step" which never felt quite right, so I changed it to "latest". Everything naturally revolves around which step was run most recently, since they build on each other. Do you prefer another name?